pkg/gitprovider/bitbucketserver.go

```diff
@@ -0,0 +1,374 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package gitprovider
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	bitbucketv1 "github.com/gfleury/go-bitbucket-v1"
+	"github.com/mitchellh/mapstructure"
+)
+
+type BitbucketServerGitProvider struct {
+	*AbstractGitProvider
+
+	username   string
+	token      string
+	baseApiUrl *string
+}
+
+const bitbucketServerResponseLimit = 100
+
+func NewBitbucketServerGitProvider(username string, token string, baseApiUrl *string) *BitbucketServerGitProvider {
+	provider := &BitbucketServerGitProvider{
+		username:            username,
+		token:               token,
+		AbstractGitProvider: &AbstractGitProvider{},
+		baseApiUrl:          baseApiUrl,
+	}
+	provider.AbstractGitProvider.GitProvider = provider
+
+	return provider
+}
+
+func (g *BitbucketServerGitProvider) getApiClient() interface{} {
```

Comment:
I suggest that you do the object cast in this method and make it return an error as well.

```golang
// Bitbucket Data Center/Server
	bitbucketDCClient, ok := client.(*bitbucketv1.APIClient)
	if !ok {
		return nil, fmt.Errorf("Invalid Bitbucket Data Center/Server client")
	}
```

It's not ideal to cast the object in every single other method.

Reply:
done..